src/common/src/block_builder.cc

```diff
@@ -0,0 +1,147 @@
+#include "block_builder.h"
+#include <bits/stdc++.h>
+
+BlockBuilder::BlockBuilder(void)
+{
+}
+
+BlockBuilder::BlockBuilder(const std::vector<std::vector<std::string> > &records)
+{
+    for (size_t i = 1; i < records.size(); ++i)
+    {
+        const auto&  record = records[i];
+        types::Block block  = ConvertRecordToBlock(record);
+        blocks_.push_back(block);
+    }
+}
+
+void AssignBlockInfrastructure(types::Block &block, const std::string &infrastructure)
+{
+    std::string infra = infrastructure;
+    std::transform(infra.begin(), infra.end(), infra.begin(), ::tolower);
+    if (infra.find("station") != std::string::npos)
+    {
+        block.has_station  = true;
+        block.station_name = infra;
```

Comment:
In the case that it has a station, is the entire `infra` string the station name?  Not just a substring of `infra`?  If that is the case, should we denote this with `if/else` statements?